tap_dynamics/sync.py

```diff
@@ -32,8 +35,15 @@ def sync(config, config_path, state, catalog):
                 stream.replication_key
             )
 
-            state = stream_obj.sync(state, stream_schema, stream_metadata, config, transformer)
+            state = stream_obj.sync(state, stream_schema, stream_metadata, config, transformer, desired_columns)
             singer.write_state(state)
 
     state = singer.set_currently_syncing(state, None)
     singer.write_state(state)
+
+
+def should_sync_column(metadata, field_name):
+    field_metadata = metadata.get(('properties', field_name), {})
+    return singer.should_sync_field(field_metadata.get('inclusion'),
+                                    field_metadata.get('selected'),
+                                    True)
```

Comment:
source code for singer.should_sync_field:
![image](https://user-images.githubusercontent.com/86244437/222761634-0463173d-9f3a-42b9-af32-c63f79f0de8f.png)